macros/edr/dbt_artifacts/upload_dbt_exposures.sql

```diff
@@ -52,5 +54,6 @@
         'path': node_dict.get('path'),
         'generated_at': elementary.datetime_now_utc_as_string()
       }%}
+    {% do flatten_exposure_metadata_dict.update({"artifact_hash": elementary.get_artifact_hash(flatten_exposure_metadata_dict)}) %}
```

Comment:
Why not doing it in the dict above? (all the comments here are relevant for all the different artifacts)

Reply:
Because it's a dict literal, you can't access properties of a dict within the dict.

```py
In [118]: d = {
     ...: 'a': 1,
     ...: 'b': d['a'] + 1
     ...: }
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
Cell In[118], line 3
      1 d = {
      2 'a': 1,
----> 3 'b': d['a'] + 1
      4 }

NameError: name 'd' is not defined

```